Doc/glossary.rst

```diff
@@ -921,6 +927,11 @@ Glossary
       reserved for rare cases where there are large numbers of instances in a
       memory-critical application.
 
+   set comprehension
+      A compact way to process all or part of the elements in a sequence and
+      return a set with the results. ``results = {'foo' for _ in range(10)}``
```

Comment:
@serhiy-storchaka thanks for mention, change another example for set comprehension.